src/contract_client.rs

```diff
@@ -832,19 +1297,14 @@ impl<Type> ContractClient<Type> {
             message,
         };
 
-        let context = ContractContext::new_from_payload(sender, None, payload);
+        let context = ContractContext::new_from_payload(sender, None, payload.clone());
 
         let invoke_result = self
             .client
             .invoke_instance(BlockIdentifier::LastFinal, &context)
             .await?
             .response;
-        let payload = UpdateContractPayload {
-            amount,
-            address: context.contract,
-            receive_name: context.method,
-            message: context.parameter,
-        };
+
```

Comment:
I don't think we should change this.
This change reduces the code, but results in extra allocations. The previous solution was reusing the same allocations.